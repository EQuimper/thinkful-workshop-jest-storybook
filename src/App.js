import React, { Component } from 'react';
import styled from 'styled-components';

import List from './List';
import Card from './components/Card'
import logo from './logo.svg';
import './App.css';

export const Title = styled.h2`
  font-size: 20;
  color: ${({ color }) => color || 'black'};
`

class App extends Component {
  render() {
    return (
      <div className="App">
        <img src={logo} className="App-logo" alt="logo" />
        <Title>Welcome to Thinkful</Title>
        <Title color="blue">Welcome to React</Title>
        <Card>
          <Title color="red">Welcome to React</Title>
        </Card>
        <List data={['todo1', 'todo2', 'todo3']} />
      </div>
    );
  }
}

export default App;
