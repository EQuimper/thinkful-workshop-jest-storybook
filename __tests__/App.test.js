import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from '../src/App';

describe('<App />', () => {
  const Component = shallow(<App />);
  it('create an basic snapshot', () => {
    const component = renderer.create(<App />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('have an wrapper className of App', () => {
    expect(Component.find('div').hasClass('App')).toBe(true);
  });
});
