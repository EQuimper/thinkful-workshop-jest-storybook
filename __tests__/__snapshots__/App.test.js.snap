exports[`<App /> create an basic snapshot 1`] = `
<div
  className="App">
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg" />
  <h2
    className="dCMMWH">
    Welcome to Thinkful
  </h2>
  <h2
    className="nIfoE"
    color="blue">
    Welcome to React
  </h2>
  <div
    className="eWbGRP">
    <h2
      className="bUfWaj"
      color="red">
      Welcome to React
    </h2>
  </div>
  <div>
    <li
      className="idnyMJ">
      todo1
    </li>
    <li
      className="idnyMJ">
      todo2
    </li>
    <li
      className="idnyMJ">
      todo3
    </li>
  </div>
</div>
`;
